lib/lz4.c

```diff
@@ -369,7 +372,15 @@ LZ4_memcpy_using_offset(BYTE* dstPtr, const BYTE* srcPtr, BYTE* dstEnd, const si
     BYTE v[8];
     switch(offset) {
     case 1:
+#if defined(__aarch64__) && defined(__GNUC__)
+        v[0] = srcPtr[0];
+        v[1] = srcPtr[0];
+        v[2] = srcPtr[0];
+        v[3] = srcPtr[0];
+        memcpy(v+4, v, 4);
```

Comment:
This modification is interesting.
It's correct I think, just it's surprising it ends up being faster than `memset()`.
I guess it might depend on the compiler, and its ability to properly optimize `memset()` for a constant length.
Anyway, it would be good to try to isolate how much gains are achieved through this change alone.

Reply:
Hi @Cyan4973 :
Please refer to below dump from ARM64:
The GCC invoke memset API directlt.
    case 1:
        memset(v, *srcPtr, 8);
     390:	f94013e0 	ldr	x0, [sp,#32]
     394:	39400000 	ldrb	w0, [x0]
     398:	2a0003e1 	mov	w1, w0
     39c:	9100e3e0 	add	x0, sp, #0x38
     3a0:	d2800102 	mov	x2, #0x8                   	// #8
     3a4:	94000000 	bl	0 <memset>
It is need more time from gprof:
  2.22      2.14     0.05                             memset
On the other hand, X86 GCC invoke the memset with avx instruction which provide much better performance than generic implementation.
  1.05      0.94     0.01                             __memset_avx2